internal/compatibility/proxy_restart.go

```diff
@@ -0,0 +1,54 @@
+package compatibility
+
+import (
+	"context"
+	"github.com/kyma-project/istio/operator/api/v1alpha2"
+	"github.com/kyma-project/istio/operator/internal/filter"
+	"github.com/kyma-project/istio/operator/internal/reconciliations/istio"
+	v1 "k8s.io/api/core/v1"
+)
+
+type ProxyRestartPredicate struct {
+	istioCR *v1alpha2.Istio
+	config  config
+}
+
+func NewRestartPredicate(istioCR *v1alpha2.Istio) *ProxyRestartPredicate {
+	return &ProxyRestartPredicate{istioCR: istioCR, config: config{proxyMetadata: v1alpha2.ProxyMetaDataCompatibility}}
+}
+
+type config struct {
+	proxyMetadata map[string]string
+}
+
+func (c config) hasProxyMetadata() bool {
+	return len(c.proxyMetadata) > 0
+}
+
+func (p ProxyRestartPredicate) NewProxyRestartEvaluator(_ context.Context) (filter.ProxyRestartEvaluator, error) {
```

Comment:
This parameter is defined in the function `NewProxyRestartEvaluator` in the `SidecarProxyPredicate` interface.
I was thinking about `SidecarProxyPredicate` and it might be possible to remove that completely for now.

Reply:
do we want to think about it in the scope of this PR, or as a technical debt followup?

Reply:
I would handle it in a follow up.